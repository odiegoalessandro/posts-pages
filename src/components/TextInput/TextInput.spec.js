import { render, screen } from "@testing-library/react";
import { TextInput } from ".";

describe('<TextInput />', () => {
  it('should have a value of searchValue', () => {
    const fn = jest.fn();
    render(<TextInput onChange={fn} searchValue={'testando...'} />);
  
    const input = screen.getByPlaceholderText(/search here/i);
    expect(input.getAttribute('placeholder')).toBe("Search here");
  });

  // it('should call handleChange function on each key pressed', () => {
  //   const fn = jest.fn();
  //   render(<TextInput handleChange={fn} />);

  //   const input = screen.getByPlaceholderText(/Search here/i);
  //   const value = 'value';

  //   userEvent.type(input, value);

  //   expect(input.value).toBe(value);
  //   expect(fn).toHaveBeenCalledTimes(value.length;
  // });

  it('should match snapshot', () => {
    const fn = jest.fn();
    const { asFragment } = render(<TextInput handleChange={fn} />);
  
    expect(asFragment()).toMatchSnapshot();
  });
});