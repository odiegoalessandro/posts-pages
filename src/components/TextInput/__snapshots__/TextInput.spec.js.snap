// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match snapshot 1`] = `
<DocumentFragment>
  <input
    class="input"
    placeholder="Search here"
    type="search"
    value=""
  />
</DocumentFragment>
`;
